### ISSUE TITLE
onevm ignoring kernel parameters defined in vm template

### ISSUE THREAD

**Issue by odairmario** [OpenNebula/one#4893]
**Description**
When i try deploy a vm with `onevm deploy <vm name> <host>` onevm generate deployment.0 file without the kernel parameters, like KERNEL_DS or KERNEL_CMD.  

I tried with this vm template
```bash
TEMPLATE 295 INFORMATION                                                        
ID             : 295                 
NAME           : test          
USER           : oneadmin            
GROUP          : oneadmin            
LOCK           : None                
REGISTER TIME  : 06/03 13:07:25      

PERMISSIONS                                                                     
OWNER          : um-                 
GROUP          : ---                 
OTHER          : ---                 

TEMPLATE CONTENTS                                                               
CPU="0.5"
CPU_MODEL=[
 MODEL="host-passthrough" ]
DISK=[
 IMAGE_ID="446" ]
FEATURES=[
 ACPI="yes",
 APIC="yes",
 PAE="no" ]
GRAPHICS=[
 LISTEN="0.0.0.0",
 TYPE="VNC" ]
MEMORY="1024"
MEMORY_UNIT_COST="MB"
NIC=[
 MAC="omitted",
 MODEL="virtio",
 NETWORK="omitted",
 NETWORK_UNAME="omitted" ]
OS=[
 ARCH="x86_64",
 KERNEL_CMD="console=ttyS0,115200n8 console=tty0",
 KERNEL_DS="$FILE[IMAGE_ID=176]",
 ROOT="vda" ]
RAW=[
 DATA="<devices> <serial type='pty'> <target port='0'/> </serial> <console type='pty'> <target type='serial' port='0'/> </console> </devices>",
 TYPE="kvm" ]
SCHED_REQUIREMENTS="ID=\"1\""
VCPU="2"
```

These following parameters are not defined in deployment.0 file:

```
 KERNEL_CMD="console=ttyS0,115200n8 console=tty0",
 KERNEL_DS="$FILE[IMAGE_ID=176]",
```

`onevm deploy test <host>` generated this deployment.0 file:
```xml
<domain type='kvm' xmlns:qemu='http://libvirt.org/schemas/domain/qemu/1.0'>
        <name>one-1446</name>
        <title>test</title>
        <vcpu><![CDATA[2]]></vcpu>
        <cputune>
                <shares>512</shares>
        </cputune>
        <memory>1048576</memory>
        <os>
                <type arch='x86_64'>hvm</type>
        </os>
        <cpu mode='host-passthrough'>
        </cpu>
        <devices>
                <emulator><![CDATA[/usr/local/bin/qemu-system-x86_64]]></emulator>
                <disk type='network' device='disk'>
                        <source protocol='gluster' name='omitted>
                                <host name='omitted' port='omitted' transport='tcp'/>
                        </source>
                        <target dev='vda' bus='virtio'/>
                        <driver name='qemu' type='raw' cache='none' io='native'/>
                </disk>
                <interface type='bridge'>
                        <source bridge='omitted'/>
                        <mac address='omitted'/>
                        <target dev='one-1446-0'/>
                        <model type='virtio'/>                </interface>
                <graphics type='vnc' listen='0.0.0.0' port='7346'/>
        </devices>
        <features>
                <acpi/>
                <apic/>
        </features>
        <devices> <serial type='pty'> <target port='0'/> </serial> <console type='pty'> <target type='serial' port='0'/> </console> </devices>
        <metadata>
                <one:vm xmlns:one="http://opennebula.org/xmlns/libvirt/1.0">
                        <one:system_datastore><![CDATA[/var/lib/one//datastores/101/1446]]></one:system_datastore>
                        <one:name><![CDATA[teste]]></one:name>
                        <one:uname><![CDATA[oneadmin]]></one:uname>
                        <one:uid>0</one:uid>
                        <one:gname><![CDATA[oneadmin]]></one:gname>
                        <one:gid>0</one:gid>
                        <one:opennebula_version>5.10.1</one:opennebula_version>
                        <one:stime>omitted</one:stime>
                        <one:deployment_time>omitted</one:deployment_time>
                </one:vm>
        </metadata>
</domain>

```
In deployment.0 or xml generated, section `<os>` contains only 
```xml
  <os>
                <type arch='x86_64'>hvm</type>
  </os>
```
instead:
```xml
<os>
    <type arch='x86_64'>hvm</type>
    <kernel>/var/lib/one//datastores/101/<id>/kernel</kernel>
    <cmdline>root=/dev/vda console=ttyS0,115200n8 console=tty0</cmdline>
</os>
```
**To Reproduce**
 1 - install or update open nebula to version 5.10.1
 2 - create a template with custom kernel, specificating  kernel file with `KERNEL_DS`
3 - create a vm with previous template and deploy with `onevm <vmname> <host>`
4 - check deployment file. 
**Expected behavior**
I expected that qemu load kernel from file defined in `KERNEL_DS` with the options defined in `KERNEL_CMD`, like this:
```xml
<os>
    <type arch='x86_64'>hvm</type>
    <kernel>/var/lib/one//datastores/101/<id>/kernel</kernel>
    <cmdline>root=/dev/vda console=ttyS0,115200n8 console=tty0</cmdline>
</os>
```
**Details**
 - Affected Component: onevm
 - Hypervisor: KVM
 - Version: 5.10.1

**Additional context**

log of onevm:

```log
/var/log/one/<id>.log
Sat Jun  6 00:17:16 2020 [Z0][VM][I]: New state is ACTIVE
Sat Jun  6 00:17:16 2020 [Z0][VM][I]: New LCM state is PROLOG
Sat Jun  6 00:17:19 2020 [Z0][VM][I]: New LCM state is BOOT
Sat Jun  6 00:17:19 2020 [Z0][VMM][I]: Generating deployment file: /var/lib/one/vms/1450/deployment.0
Sat Jun  6 00:17:19 2020 [Z0][VM][I]: Virtual Machine has no context
Sat Jun  6 00:17:19 2020 [Z0][VMM][I]: ExitCode: 0
Sat Jun  6 00:17:19 2020 [Z0][VMM][I]: Successfully execute network driver operation: pre.
Sat Jun  6 00:17:20 2020 [Z0][VMM][I]: ExitCode: 0
Sat Jun  6 00:17:20 2020 [Z0][VMM][I]: Successfully execute virtualization driver operation: deploy.
Sat Jun  6 00:17:20 2020 [Z0][VMM][I]: ExitCode: 0
Sat Jun  6 00:17:20 2020 [Z0][VMM][I]: Successfully execute network driver operation: post.
Sat Jun  6 00:17:20 2020 [Z0][VM][I]: New LCM state is RUNNING
```

My workaround to work is edit vm with `virsh edit one-<id>` and add these followings lines in `<os>` section:
```xml
    <kernel>/var/lib/one//datastores/101/<id>/kernel</kernel>
    <cmdline>root=/dev/vda console=ttyS0,115200n8 console=tty0</cmdline>
```
and run:
```bash
virsh destroy one-<id>
virsh start one-<id>
```

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by odairmario** [comment]
# update
I followed this post https://forum.opennebula.io/t/kernel-ds-image-doesnt-appear-in-the-qemu-system-x86-64-command/7569/5 to tried solve the problem, but didnt solve.

**Comment by christian7007** [comment]
Hello @odairmario, the fix mentioned in the forum comment was added in OpenNebula 5.10.4. Did you try it in that version too?

**Comment by odairmario** [comment]
resolved, when i upgrade to version 5.10.5 from source code, the binary oned not updated and continued in version 5.10.1, then i moved src/nebula/oned version 5.10.5 to /usr/bin/oned and work. 
ty for attention. 

## Possible bug: i executed this command line and not copied  new oned binary to /usr/bin/oned
```bash
./install.sh -u oneadmin -g oneadmin -k -s -c cli
```
maybe i did something wrong to not copy.

**Comment by christian7007** [comment]
I think the problem is that you used the `-c` flag:

```
-c | only install client utilities: OpenNebula cli and ec2 client files
```
